peer.go

```diff
@@ -63,6 +64,64 @@ func (id ID) MatchesPublicKey(pk ic.PubKey) bool {
 	return oid == id
 }
 
+func (id ID) ExtractEd25519PublicKey() ic.PubKey {
+	// ed25519 pubkey identity format
+	// <identity mc><length (2 + 32 = 34)><ed25519-pub mc><ed25519 pubkey>
+	// <0x00       ><0x22                ><0xed01        ><ed25519 pubkey>
```

Comment:
Ok. It seems the documentation is correct with `0xed01` (see [code here](https://play.golang.org/p/12Lw8AOn1l)). Could you confirm this? And could you expand on why you think that the length might be off in [this comment](https://github.com/libp2p/go-libp2p-peer/pull/13#discussion_r116257739).